### ISSUE TITLE
Enabling the asset pipeline breaks svg paths

### ISSUE THREAD

**Issue by butonic** [owncloud/core#24873]
### Steps to reproduce
1. enable asset pipeline
3. open dev console with f12
2. open the files app

### Expected behaviour
no errors

### Actual behaviour
![assets-svg](https://cloud.githubusercontent.com/assets/956847/15607316/210884d4-2412-11e6-9f54-0cffbd86cbd9.PNG)
It seems two characters are truncated.

### Server configuration**Operating System:**
debian 7.10

**Web Server:**
Server version: Apache/2.2.22 (Debian)
Server built:   Aug 18 2015 09:49:50

**Database:**
mysql  Ver 14.14 Distrib 5.5.49, for debian-linux-gnu (x86_64) using readline 6.2

**PHP version:**
PHP 5.6.21-1~dotdeb+7.1 (cli) (built: Apr 29 2016 13:07:22) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies

**ownCloud version:**
  - version: 9.0.2.2

**Updated from an older ownCloud or fresh install:**
updated from 8.2.5

**Where did you install ownCloud from:**
tar.bz

**Signing status (ownCloud 9.0 and above):**
```
```

**List of activated apps:***

```
Enabled:
  - activity: 2.2.1
  - bookmarks: 0.7
  - calendar: 1.2.2
  - comments: 0.2
  - contacts: 1.2.0.0
  - dav: 0.1.6
  - direct_menu: 0.7.1
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - galleryplus: 15.0.0
  - guests: 0.0.4
  - logreader: 1.1.2
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - search_elastic: 0.0.10
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_external
  - files_mv
  - gallery
  - metarename
  - user_external
  - user_ldap
  - weather
```

**The content of config/config.php:**

```
{
    "system": {
        "datadirectory": "\/home\/jfd\/owncloud\/data",
        "dbtype": "mysql",
        "version": "9.0.2.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 0,
        "lastupdatedat": "1333128672.0066",
        "instanceid": "4ff312b01bc78",
        "ldapIgnoreNamingRules": false,
        "maintenance": false,
        "forcessl": true,
        "filesystem_check_changes": 0,
        "theme": "",
        "apps_paths": [
            {
                "path": "\/var\/www\/owncloud\/htdocs\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/owncloud\/htdocs\/apps-repos",
                "url": "\/apps-repos",
                "writable": true
            }
        ],
        "trusted_domains": [
            "domain.tld"
        ],
        "secret": "***REMOVED SENSITIVE VALUE***",
        "appstore.experimental.enabled": true,
        "singleuser": false,
        "trashbin_retention_obligation": "auto",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0
        },
        "overwrite.cli.url": "https:\/\/domain.tld",
        "asset-pipeline.enabled": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\Movie"
        ]
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Client configuration
**Browser:**
Chromium
**Operating system:**
Windows 10
#### Web server error log
```
[Fri May 27 14:24:10 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:24:10 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/stemtags
[Fri May 27 14:24:10 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:24:10 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:27:30 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:27:30 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:27:30 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:27:30 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
[Fri May 27 14:27:30 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/stemtags
[Fri May 27 14:27:30 2016] [error] [client 77.21.197.122] File does not exist: /var/www/owncloud/htdocs/les
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Run job with ID 6","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Finished job with ID 6","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Run job with ID 5","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Finished job with ID 5","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Run job with ID 4","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Finished job with ID 4","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Run job with ID 3","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Finished job with ID 3","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Run job with ID 2","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
{"reqId":"7M\/SUEjKTWEa3P7w1mmM","remoteAddr":"","app":"cron","message":"Finished job with ID 2","level":0,"time":"2016-05-27T12:15:01+00:00","method":"--","url":"--","user":"--"}
```

#### Browser log
```
see screenshot above
```

**Comment by butonic** [comment]
the generated css actually has the wrong path:
```
assets/6006e725629a2961c76f09724ba702bc.css:[...].nav-icon-files{background-image:url('../les/img/folder.svg')}
```

**Comment by butonic** [comment]
deleting the assets regenerates them with broken path

**Comment by butonic** [comment]
my instance seems to not use pretty urls: `https://oc.domain.tld/index.php/apps/files/` maybe this is not working as expected:
```
		if (!file_exists("$assetDir/assets/$cssHash.css")) {
			$cssFiles = array_map(function ($item) {
				$root = $item[0];
				$file = $item[2];
				$assetPath = $root . '/' . $file;
```

**Comment by DeepDiver1975** [comment]
asset pipeline works here with and without pretty urls ...

@butonic mind sharing your htaccess?

**Comment by butonic** [comment]
I updated from 8.2.3 to 9.0.2 and it seems the .htaccess was not updated correctly. It was lacking the end after `#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####`:
```
# line below if for Apache 2.4
<ifModule mod_authz_core.c>
Require all denied
</ifModule>

# line below if for Apache 2.2
<ifModule !mod_authz_core.c>
deny from all
</ifModule>

# section for Apache 2.2 and 2.4
IndexIgnore *
root@butonic:/var/www/owncloud/htdocs/config# cd ..
root@butonic:/var/www/owncloud/htdocs# cat .htaccess 
<IfModule mod_headers.c>
  <IfModule mod_setenvif.c>
    <IfModule mod_fcgid.c>
       SetEnvIfNoCase ^Authorization$ "(.+)" XAUTHORIZATION=$1
       RequestHeader set XAuthorization %{XAUTHORIZATION}e env=XAUTHORIZATION
    </IfModule>
    <IfModule mod_proxy_fcgi.c>
       SetEnvIfNoCase Authorization "(.+)" HTTP_AUTHORIZATION=$1
    </IfModule>
  </IfModule>

  <IfModule mod_env.c>
    # Add security and privacy related headers
    Header set X-Content-Type-Options "nosniff"
    Header set X-XSS-Protection "1; mode=block"
    Header set X-Robots-Tag "none"
    Header set X-Frame-Options "SAMEORIGIN"
    Header set X-Download-Options "noopen"
    Header set X-Permitted-Cross-Domain-Policies "none"
    SetEnv modHeadersAvailable true
  </IfModule>

  # Add cache control for CSS and JS files
  <FilesMatch "\.(css|js)$">
    Header set Cache-Control "max-age=7200, public"
  </FilesMatch>
</IfModule>
<IfModule mod_php5.c>
  php_value upload_max_filesize 513M
  php_value post_max_size 513M
  php_value memory_limit 512M
  php_value mbstring.func_overload 0
  php_value always_populate_raw_post_data -1
  php_value default_charset 'UTF-8'
  php_value output_buffering 0
  <IfModule mod_env.c>
    SetEnv htaccessWorking true
  </IfModule>
</IfModule>
<IfModule mod_php7.c>
  php_value upload_max_filesize 513M
  php_value post_max_size 513M
  php_value memory_limit 512M
  php_value mbstring.func_overload 0
  php_value default_charset 'UTF-8'
  php_value output_buffering 0
  <IfModule mod_env.c>
    SetEnv htaccessWorking true
  </IfModule>
</IfModule>
<IfModule mod_rewrite.c>
  RewriteEngine on
  RewriteRule .* - [env=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
  RewriteRule ^\.well-known/host-meta /public.php?service=host-meta [QSA,L]
  RewriteRule ^\.well-known/host-meta\.json /public.php?service=host-meta-json [QSA,L]
  RewriteRule ^\.well-known/carddav /remote.php/dav/ [R=301,L]
  RewriteRule ^\.well-known/caldav /remote.php/dav/ [R=301,L]
  RewriteRule ^remote/(.*) remote.php [QSA,L]
  RewriteRule ^(build|tests|config|lib|3rdparty|templates)/.* - [R=404,L]
  RewriteCond %{REQUEST_URI} !^/.well-known/acme-challenge/.*
  RewriteRule ^(\.|autotest|occ|issue|indie|db_|console).* - [R=404,L]

  # Rewrite rules for `front_controller_active`
  Options -MultiViews
  RewriteRule ^core/js/oc.js$ index.php [PT,E=PATH_INFO:$1]
  RewriteRule ^core/preview.png$ index.php [PT,E=PATH_INFO:$1]
  RewriteCond %{REQUEST_FILENAME} !\.(css|js|svg|gif|png|html|ttf|woff|ico|jpg|jpeg)$
  RewriteCond %{REQUEST_FILENAME} !core/img/favicon.ico$
  RewriteCond %{REQUEST_FILENAME} !/remote.php
  RewriteCond %{REQUEST_FILENAME} !/public.php
  RewriteCond %{REQUEST_FILENAME} !/cron.php
  RewriteCond %{REQUEST_FILENAME} !/core/ajax/update.php
  RewriteCond %{REQUEST_FILENAME} !/status.php
  RewriteCond %{REQUEST_FILENAME} !/ocs/v1.php
  RewriteCond %{REQUEST_FILENAME} !/ocs/v2.php
  RewriteCond %{REQUEST_FILENAME} !/updater/
  RewriteCond %{REQUEST_FILENAME} !/ocs-provider/
  RewriteCond %{REQUEST_URI} !^/.well-known/acme-challenge/.*
</IfModule>
<IfModule mod_mime.c>
  AddType image/svg+xml svg svgz
  AddEncoding gzip svgz
</IfModule>
<IfModule mod_dir.c>
  DirectoryIndex index.php index.html
</IfModule>
AddDefaultCharset utf-8
Options -Indexes
<IfModule pagespeed_module>
  ModPagespeed Off
</IfModule>

#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####

ErrorDocument 403 /core/templates/403.php
ErrorDocument 404 /core/templates/404.php
<IfModule mod_rewrite.c>
  RewriteRule . index.php [PT,E=PATH_INFO:$1]
  RewriteBase /
  <IfModule mod_env.c>
    SetEnv front_controller_active true
    <IfModule mod_dir.c>
      DirectorySlash off
    </IfModule>
  </IfModule>
</IfModule>
```

After adding it manually, deleting the asset dir and reenabling the asset pipeline the urls are generated correctly.

To upgrade I moved away the old sources and copied the config (as is the documented way of updating), so the htaccess was the one from the tarball. In 9 there seems to be a commad that allows checking the htaccess. And this PR should help people solve the problem should they run into it. So ... I guess fixed in master ...

**Comment by DeepDiver1975** [comment]
> I guess fixed in master ...

close then?

**Comment by butonic** [comment]
for reference, fixed in 9.0.3: https://github.com/owncloud/core/pull/14081#issuecomment-220440352
